pgjdbc/src/main/java/com/yugabyte/ysql/ClusterAwareLoadBalancer.java

```diff
@@ -201,8 +230,16 @@ protected ArrayList<String> getCurrentServers(Connection conn) throws SQLExcepti
       String region = rs.getString("region");
       String zone = rs.getString("zone");
       hostPortMap.put(host, port);
+     if(this instanceof TopologyAwareLoadBalancer  && !unreachableHosts.containsKey(host)) {
+        int priority = getPriority(cloud, region, zone);
+        LOGGER.log(Level.FINE, "Priotity for host "
+            + host + " = " + priority);
+        hostToPriorityMap.put(host,priority);
+      }
```

Comment:
Move these in a method and put it in `TopologyAwareLoadBalancer`. It's implementation in `ClusterAwareLoadBalancer` will be a no-op. Also, we should get the priority of a node irrespective of its availability.

```
protected void updatePriorityMap() {
    int priority = getPriority(cloud, region, zone);
    LOGGER.log(Level.FINE, "Priority of host "
        + host + " = " + priority);
     hostToPriorityMap.put(host, priority);
}
```

Reply:
If we get the priority of a node irrespective of its availability, it would lead to a higher latency.